src/System.Configuration.ConfigurationManager/src/System/Configuration/Internal/WriteFileContext.cs

```diff
@@ -4,6 +4,11 @@
 
 using System.IO;
 using System.Threading;
+#if CODEDOM
+using System.CodeDom.Compiler;
+#else
+using System.IO.Internal;
+#endif
```

Comment:
@JeremyKuhne, why can't we take a dependency on System.CodeDom.Compiler in corefx here? i.e. why is this assembly building in its own copy of TempFileCollection?

Reply:
I don't recall.  Might not have been available at the time?

Reply:
I took it as a corefx housecleaning move.  But, if you guys have CodeDom as part of corefx now, it would make sense to take a dependency.

Reply:
Is there any reason not to remove `IO.Internal.TempFileCollection` at this time then? It sounds like @steveisok you don't want it either.

Reply:
@weshaggard can you sanity check? Are we good to take the dependency now?

Reply:
Ultimately, either taking a dependency or merging this PR will accomplish my goal.  I'm good either way.

Reply:
CodeDom is not part of the shared framework and so I would be shy at taking a dependency on it here, especially just for the small TempFileCollection type. We could however consider moving that type down into lower level library.

Reply:
It isn't yet, but even as packages I think depending on all of CodeDom for this one type seems like a big dependency for small gain.

Reply:
OK, so could this PR move forward as somewhat of a stop-gap?

Reply:
> OK, so could this PR move forward as somewhat of a stop-gap?

I'm fine with this as a work around.